### ISSUE TITLE
not enough space in the context's memory pool

### ISSUE THREAD

**Issue by GC-Jacky** [imartinez/privateGPT#104]
Hi I'm using a Mac M1 16G. 
Sometimes when I entered a query, the script crash as below:

```
[2023-05-14 13:48:12,141] {chroma.py:128} ERROR - Chroma collection langchain contains fewer than 4 elements.
[2023-05-14 13:48:12,142] {chroma.py:128} ERROR - Chroma collection langchain contains fewer than 3 elements.
[2023-05-14 13:48:12,142] {chroma.py:128} ERROR - Chroma collection langchain contains fewer than 2 elements.
ggml_new_tensor_impl: not enough space in the context's memory pool (needed 5246435536, available 5243946400)
[1]    20822 segmentation fault  python privateGPT.py
```

Anyone knows why?

**Comment by lu161513** [comment]
I also face this situation, how can I change the memory_size for `gptj_model_load: memory_size =   896.00 MB, n_mem = 57344`

**Comment by Swader** [comment]
Same issue, M1 Mac with 32 GB RAM.

**Comment by ReplicantArmy** [comment]
Same issues, fails on 2nd query each time, tested on 128gb ram desktop and 64gb laptop

**Comment by sbnarra** [comment]
Seeing this same issue, have 32gb ram.

Before asking any questions I see memory jump by about 8gb (loading the model/embeddings).

Then with each question memory is going up by anywhere between 2/20gb before dying with 

```
llama_print_timings:        load time =   462.49 ms
llama_print_timings:      sample time =     0.00 ms /     1 runs   (    0.00 ms per run)
llama_print_timings: prompt eval time =   500.75 ms /     8 tokens (   62.59 ms per token)
llama_print_timings:        eval time =     0.00 ms /     1 runs   (    0.00 ms per run)
llama_print_timings:       total time =   502.85 ms
gpt_tokenize: unknown token '�'
gpt_tokenize: unknown token '�'
gpt_tokenize: unknown token '�'
gpt_tokenize: unknown token '�'
gpt_tokenize: unknown token '�'
gpt_tokenize: unknown token '�'
gpt_tokenize: unknown token '�'
gpt_tokenize: unknown token '�'
gpt_tokenize: unknown token '�'
gpt_tokenize: unknown token '�'
gpt_tokenize: unknown token '�'
gpt_tokenize: unknown token '�'
ggml_new_tensor_impl: not enough space in the context's memory pool (needed 8608229136, available 8598029008)
[1]    128505 segmentation fault (core dumped)  
```

tried playing around with smaller context sizes but memory seems to keep increasing till out of mem.

**Comment by d2rgaming-9000** [comment]
Same issue #141

**Comment by d2rgaming-9000** [comment]
I just have a hunch that it's got something to do with invoking the documents, especially if it's a very large DB.

**Comment by d2rgaming-9000** [comment]
We need to set a variable that checks the max memory used, and string whatever's left as stream [until more memory is freed] instead of streaming everying at once, wherein it crashes.

**Comment by hatgit** [comment]
I am getting this error too: `ggml_new_tensor_impl: not enough space in the context's memory pool (needed 20923687696, available 20754450208)`
